src/Adapter/CustomerService/QueryHandler/GetCustomerThreadForViewingHandler.php

```diff
@@ -0,0 +1,386 @@
+            }
+        }
+
+        krsort($timeline);
+
+        $timelineItems = [];
+
+        foreach ($timeline as $items) {
+            foreach ($items as $item) {
+                $timelineItems[] = new CustomerThreadTimelineItem(
+                    $item['content'],
+                    $item['icon'],
+                    $item['arrow'],
+                    $item['date'],
+                    isset($item['background_color']) ? $item['background_color'] : null
+                );
+            }
+        }
+
+        return new CustomerThreadTimeline($timelineItems);
+    }
+
+    /**
+     * @param CustomerThread $thread
+     *
+     * @return array
+     */
+    private function getAvailableActions(CustomerThread $thread)
+    {
+        $actions = [];
+
+        if ($thread->status !== 'closed') {
+            $actions[CustomerThreadStatus::CLOSED] = [
+                'label' => $this->translator->trans('Mark as "handled"', [], 'Admin.Catalog.Feature'),
+                'value' => CustomerThreadStatus::CLOSED,
+            ];
+        } else {
+            $actions[CustomerThreadStatus::OPEN] = [
+                'label' => $this->translator->trans('Re-open', [], 'Admin.Catalog.Feature'),
+                'value' => CustomerThreadStatus::OPEN,
+            ];
+        }
+
+        if ($thread->status !== 'pending1') {
+            $actions[CustomerThreadStatus::PENDING_1] = [
+                'label' => $this->translator->trans(
+                    'Mark as "pending 1" (will be answered later)',
+                    [],
+                    'Admin.Catalog.Feature'
+                ),
+                'value' => CustomerThreadStatus::PENDING_1,
+            ];
+        } else {
+            $actions[CustomerThreadStatus::PENDING_1] = [
+                'label' => $this->translator->trans('Disable pending status', [], 'Admin.Catalog.Feature'),
+                'value' => CustomerThreadStatus::OPEN,
+            ];
+        }
+
+        if ($thread->status !== 'pending2') {
+            $actions[CustomerThreadStatus::PENDING_2] = [
+                'label' => $this->translator->trans(
+                    'Mark as "pending 2" (will be answered later)',
+                    [],
+                    'Admin.Catalog.Feature'
+                ),
+                'value' => CustomerThreadStatus::PENDING_2,
+            ];
+        } else {
+            $actions[CustomerThreadStatus::PENDING_2] = [
+                'label' => $this->translator->trans('Disable pending status', [], 'Admin.Catalog.Feature'),
+                'value' => CustomerThreadStatus::OPEN,
+            ];
+        }
+
+        return $actions;
+    }
+
+    /**
+     * @param CustomerThread $thread
+     *
+     * @return CustomerInformation
+     */
+    private function getCustomerInformation(CustomerThread $thread)
+    {
+        if (!$thread->id_customer) {
+            return new CustomerInformation(null, null, null, $thread->email, null, null, null);
```

Comment:
😆

Reply:
:sweat_smile: 

Maybe `CustomerInformation::withEmailOnly($email)`?

Reply:
Great idea ❤️

Reply:
Done.

Reply:
```CustomerInformation::with(['email' => $email]);```
Otherwise create `withLastNameOnly`, `withFirstNameOnly`, etc.

Reply:
No, I think this would be wrong, passing unstructured data like `array` does not make sense as you can pass more arguments that this Dto supports. 

In addition, `CustomerInformation` Dto is **not valid** without email, but **may not** have other information and this is perfectly fine, if our use case where to evolve, then we can think about adding other methods (e.g. `withLastNameOnly`), but only when we actually need them.